agreement-service.js

```diff
@@ -23,7 +23,13 @@ function factory($http, config) {
     });
     // TODO: make url configurable
     var url = config.data.baseUri + '/agreements/accepted';
-    return Promise.resolve($http.post(url, {agreement: agreementUrls}));
+    return Promise.resolve($http.post(url, {agreement: agreementUrls}))
+      .then(function(response) {
+        if(response.status !== 201) {
+          throw new Error('Agreement acceptance was not properly recorded.');
```

Comment:
The status check should probably be: `if(response.status >= 300)` ... we may, in the future, start using 200 instead of 201, for example, and we shouldn't need to change this code here.